### PR TITLE
fix: properly store timestamps in versions

### PR DESCRIPTION
**Pull request by r1tsuu**
This PR makes a more clear gap between `version_createdAt` / `version_updatedAt` and `createdAt` / `updatedAt` columns / fields in mongodb. 

- `createdAt` - This should be a new value in a new version. Before this change it was the same all the time. Should remain the same on autosave.
- The same for `updatedAt`, but it should be updated on every change (including autosave)
- `version_createdAt` - Should remain equal to `createdAt` from the parent collection / table
- `version_updatedAt` - On a latest version it makes sense this be the same as `updatedAt` from the parent collection / table, as all the `version_*` fields should be just synced with it

### CODE REVIEW

File: packages/ui/src/providers/DocumentInfo/index.tsx [hunk 1]
```diff
@@ -339,6 +339,11 @@ const DocumentInfo: React.FC<
               ...versionParams.where,
               and: [
                 ...versionParams.where.and,
+                {
+                  'version._status': {
+                    equals: 'draft',
+                  },
+                },
```
Comment (DanRibbens) [inline packages/ui/src/providers/DocumentInfo/index.tsx:346]: Why is this needed? I expected that we want to include published also.

**Review by DanRibbens — CHANGES_REQUESTED** [review]
Just have the one question about the front end.

File: packages/ui/src/providers/DocumentInfo/index.tsx (hunk 1 above)
Comment (r1tsuu) [reply packages/ui/src/providers/DocumentInfo/index.tsx:346]: This is used for this indicator:
image

https://github.com/payloadcms/payload/blob/067d353cdd2869623a2361322625b685ff355f5e/packages/ui/src/elements/Status/index.tsx#L42-L51   

The logic is that if there are newer versions - this should be be marked as "Changed", I.e currently this doc has status `'published'` but there is a new version that's has status `'draft``, either from "Save Draft" or autosave which creates a draft version.

We relied on a hack in `saveVersion` to have the same `updatedAt` with `doc.updatedAt` on the version itself so then this particular query doesn't return any docs when publishing.
https://github.com/payloadcms/payload/pull/8646/files#diff-e47751f12d8e80d3d391d2d4ab2a7d6739bdaf4adc3755674d045180d2741492L117-L122
But from the database perspective, it's not true that this field should have the same `updatedAt`, it should only apply for `version_updatedAt`. Imagine we updated the row in `posts` table, but the version in `_posts_v` is created 20 seconds later because of some lag.

In also this creates a bad situation for sorting by `updatedAt` in SQL databases because you can have different versions with the same `updatedAt` value.

Comment (DanRibbens) [reply packages/ui/src/providers/DocumentInfo/index.tsx:346]: That makes sense. I didn't look at how this was being used and the variables in the component aren't helping.

**Review by DanRibbens — APPROVED** [review]
Good fix!

**Comment by cbratschi** [comment]
I have a question about this change:

> - createdAt - This should be a new value in a new version. Before this change it was the same all the time. Should remain the same on autosave.
> - The same for updatedAt, but it should be updated on every change (including autosave)

We use createdAt like a published date i.e. this values should be equal to the first time the document was published. Above it is mentioned this value will change with new revisions. This will break the current behavior.

Coming from WordPress there are two dates there:

- https://developer.wordpress.org/reference/functions/get_the_date/
- https://developer.wordpress.org/reference/functions/get_the_modified_date/

There is a creation date (published date) and a modified date. If both dates are equal the article was published once without getting any update. If there are updates, the last update date is returned.

The wording "written date" might be confusing in the documentation and the following references are more precise:

https://developer.wordpress.org/reference/functions/get_post_datetime/
https://pexetothemes.com/wordpress-functions/get_the_date/

There must be a rule which separates createdAt from updatedAt.

**Comment by DanRibbens** [comment]
> I have a question about this change:
> 
> > - createdAt - This should be a new value in a new version. Before this change it was the same all the time. Should remain the same on autosave.
> > - The same for updatedAt, but it should be updated on every change (including autosave)

> We use createdAt like a published date i.e. this values should be equal to the first time the document was published. Above it is mentioned this value will change with new revisions. This will break the current behavior.

You need to make a `publishedAt` date field that you can set in a hook the first time the document has doc._status === 'published'. No date from Payload will give you exactly how you need it.
